argschema/argschema_parser.py

```diff
@@ -206,7 +207,28 @@ def output(self,d,output_path=None):
             self.logger.warning("output_schema_type is not defined,\
                                  the output won't be validated")
             output_json = d
+        
+        return output_json
+
+    def output(self,d,output_path=None):
+        """method for outputing dictionary to the output_json file path after
+        validating it through the output_schema_type
 
+        Parameters
+        ----------
+        d:dict
+            output dictionary to output 
+        output_path: str
+            path to save to output file, optional (with default to self.mod['output_json'] location)
+        Raises
+        ------
+        marshmallow.ValidationError
+            If any of the output dictionary doesn't meet the output schema
+        """
+        if output_path is None:
```

Comment:
Nifty python idiom:

output_path = self.args['output_json'] if not output_path else output_path